for-developers/walkthrough/dev-setup/making-your-first-changes.md

```diff
@@ -47,4 +47,1930 @@ Now you should see some custom content in the post that you added to the feed. Y
+			for ii := 0; ii < len(reactedPostHashes); ii++ {
+				require.Contains(m3Reacts, *reactedPostHashes[ii])
+			}
+		}
+	}
+	testConnectedState()
+
+	// Disconnect the transactions from a single view in the same way as above
+	// i.e. without flushing each time.
+	utxoView2, err := NewUtxoView(db, params, nil)
+	require.NoError(err)
+	for ii := 0; ii < len(txnOps); ii++ {
+		backwardIter := len(txnOps) - 1 - ii
+		fmt.Printf("Disconnecting transaction with index %d (going backwards)\n", backwardIter)
+		currentOps := txnOps[backwardIter]
+		currentTxn := txns[backwardIter]
+
+		currentHash := currentTxn.Hash()
+		err = utxoView2.DisconnectTransaction(currentTxn, currentHash, currentOps, savedHeight)
+		require.NoError(err)
+	}
+	require.NoError(utxoView2.FlushToDb())
+	require.Equal(expectedSenderBalances[0], _getBalance(t, chain, nil, senderPkString))
+	require.Equal(expectedRecipientBalances[0], _getBalance(t, chain, nil, recipientPkString))
+
+	testDisconnectedState()
+
+	// All the txns should be in the mempool already so mining a block should put
+	// all those transactions in it.
+	block, err := miner.MineAndProcessSingleBlock(0 /*threadIndex*/, mempool)
+	require.NoError(err)
+	// Add one for the block reward. Now we have a meaty block.
+	require.Equal(len(txnOps)+1, len(block.Txns))
+	// Estimate the transaction fees of the tip block in various ways.
+	{
+		// Threshold above what's in the block should return the default fee at all times.
+		require.Equal(int64(0), int64(chain.EstimateDefaultFeeRateNanosPerKB(.1, 0)))
+		require.Equal(int64(7), int64(chain.EstimateDefaultFeeRateNanosPerKB(.1, 7)))
+		// Threshold below what's in the block should return the max of the median
+		// and the minfee. This means with a low minfee the value returned should be
+		// higher. And with a high minfee the value returned should be equal to the
+		// fee.
+		require.Equal(int64(7), int64(chain.EstimateDefaultFeeRateNanosPerKB(0, 7)))
+		require.Equal(int64(4), int64(chain.EstimateDefaultFeeRateNanosPerKB(0, 0)))
+		require.Equal(int64(7), int64(chain.EstimateDefaultFeeRateNanosPerKB(.01, 7)))
+		require.Equal(int64(4), int64(chain.EstimateDefaultFeeRateNanosPerKB(.01, 1)))
+	}
+
+	testConnectedState()
+
+	// Roll back the block and make sure we don't hit any errors.
+	{
+		utxoView, err := NewUtxoView(db, params, nil)
+		require.NoError(err)
+
+		// Fetch the utxo operations for the block we're detaching. We need these
+		// in order to be able to detach the block.
+		hash, err := block.Header.Hash()
+		require.NoError(err)
+		utxoOps, err := GetUtxoOperationsForBlock(db, hash)
+		require.NoError(err)
+
+		// Compute the hashes for all the transactions.
+		txHashes, err := ComputeTransactionHashes(block.Txns)
+		require.NoError(err)
+		require.NoError(utxoView.DisconnectBlock(block, txHashes, utxoOps))
+
+		// Flushing the view after applying and rolling back should work.
+		require.NoError(utxoView.FlushToDb())
+	}
+
+	testDisconnectedState()
```

Comment:
I would probably use the `TestMeta` struct and follow the pattern used in `block_view_dao_coin_limit_order_test.go`. 

Using the below at the end of your tests ensures all disconnect / connects work in all different forms:

```
	_rollBackTestMetaTxnsAndFlush(testMeta)
	_applyTestMetaTxnsToMempool(testMeta)
	_applyTestMetaTxnsToViewAndFlush(testMeta)
	_disconnectTestMetaTxnsFromViewAndFlush(testMeta)
	_connectBlockThenDisconnectBlockAndFlush(testMeta)
```